From: SeongJae Park <sj@kernel.org>
Date: Thu, 23 May 2024 15:48:07 -0700
Subject: [PATCH] ==== write-only monitoring ====

Implement new monitoring primitive sets called vaddr-wo and paddr-wo.
Those are same to vaddr and paddr, but monitor only write using
write-protect faults mechanism for access checking.

On '->prepare_access_checks()', set the access sampling page as
write-protected on the page table entry (pte_wrprotect()).

If the page is written, write-protect page faults handling
(wp_huge_pud(), wp_huge_pmd(), do_wp_page()) will remove the protection
bit.

On '->check_accesses()', just check if the protection is removed
(pte_write()), consider it as accessed (written) in the case.

For possible future DAMON extensions
------------------------------------

To extend DAMON for NUMA-ware monitoring, we may reuse NUMA hinting
faults in similar way (using change_prot_numa() and pte_prot_none()).
We might better to use NUMA hinting for unitifed fault-based DAMON
extensions.

In future, DAMON could be extended for h/w-based instruction sampling.
If the h/w feature doesn't support address-based filtering, the sampled
instruction would need to find the DAMON region of the address and mark
it as accessed.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 damon_meta_changes/qLG8W05V | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 damon_meta_changes/qLG8W05V

diff --git a/damon_meta_changes/qLG8W05V b/damon_meta_changes/qLG8W05V
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

